src/foremast/s3/s3deploy.py

```diff
@@ -112,36 +120,57 @@ def promote_artifacts(self, promote_stage='latest'):
         else:
             self._sync_to_uri(self.s3_latest_uri)
 
-    def _upload_artifacts_to_version(self):
-        """Recursively upload directory contents to S3."""
+    def _get_upload_cmd(self, flat=False):
+        """Generate the S3 CLI upload command
+
+        Args:
+            flat (bool): If true, uses a flat directory structure instead of nesting under a version.
+
+        Returns:
+            str: The full CLI command to run.
+        """
+        if flat:
```

Comment:
The only difference in the `cmd` variable is that one uses `self.s3_flat_uri` vs `self.s3_version_uri`.
It would be better if it was something like:
```
if flat:
   uri = self.s3_flat
else:
  uri = self.non_s3_flat

cmd = 'aws s3 ....{} {}'.format(self.something, uri)
```

Since there's no difference in the `cmd` except for the path, this makes that more explicitly visible.

Reply:
yeah, that makes sense. I will update it